google/cloud/storage/internal/grpc/default_options_test.cc

```diff
@@ -62,12 +72,27 @@ TEST(DefaultOptionsGrpc, DefaultOptionsGrpcChannelCount) {
   }
 }
 
-TEST(DefaultOptionsGrpc, DefaultEndpoints) {
-  auto options = DefaultOptionsGrpc();
+TEST(DefaultOptionsGrpc, DefaultEndpointsCloudPath) {
+  auto mock = std::make_shared<MockGcpDetector>();
+  EXPECT_CALL(*mock, IsGoogleCloudBios()).WillRepeatedly(Return(false));
+  EXPECT_CALL(*mock, IsGoogleCloudServerless()).WillRepeatedly(Return(false));
+
+  auto options = DefaultOptionsGrpc(Options{}.set<GcpDetectorOption>(mock));
```

Comment:
consider changing the signature to:

```cc
Options DefaultOptionsGrpc(Options, std::shared_ptr<internal::GcpDetector> gcp_detector = internal::MakeGcpDetector());
```

We can still inject the mock in the unit tests.

Reply:
Similar to above, it sounds like this suggestion is due to the concerns over the option being internal currently? The whole of the gRPC implementation is internal currently.

Reply:
> The whole of the gRPC implementation is internal currently.

Uh... I am confused. The gRPC GCS Client API is experimental, not internal.

https://github.com/googleapis/google-cloud-cpp/blob/a5155607dd97c33e711d7ac382c123d7f0843c87/google/cloud/storage/grpc_plugin.h#L61

Unless you are talking about the gRPC client implementation details, which are internal, where they belong.